stake-pool/js/src/utils/stake.ts

```diff
@@ -105,21 +125,26 @@ export async function prepareWithdrawAccounts(
   const withdrawFrom: WithdrawAccount[] = [];
   let remainingAmount = amount;
 
+  const fee = stakePool.stakeWithdrawalFee;
+  const inverseFee: Fee = {
+    numerator: fee.denominator.sub(fee.numerator),
+    denominator: fee.denominator,
+  };
+
   for (const type of ['preferred', 'active', 'transient', 'reserve']) {
-    const filteredAccounts = accounts.filter(a => a.type == type);
+    const filteredAccounts = accounts.filter((a) => a.type == type);
 
-    for (const {stakeAddress, voteAddress, lamports} of filteredAccounts) {
-      let availableForWithdrawal = Math.floor(
-        calcPoolTokensForDeposit(stakePool, lamports),
-      );
-      if (!stakePool.stakeWithdrawalFee.denominator.isZero()) {
+    for (const { stakeAddress, voteAddress, lamports } of filteredAccounts) {
+      if (lamports <= minBalance) {
+        continue;
+      }
+
+      let availableForWithdrawal = calcPoolTokensForDeposit(stakePool, lamports - minBalance);
```

Comment:
@CMEONE the issue comes from this line.  Instead of always subtracting `minBalance`, it should only be subtracted from the transient accounts, as done in #2900.  The errors comes from not withdrawing everything possible from the main stake account before withdrawing from the transient stake account.

Reply:
That makes sense, but when I tried modifying that line to `let availableForWithdrawal = calcPoolTokensForDeposit(stakePool, type == 'transient' ? lamports - minBalance : lamports);`, I still got the same error `Incorrect stake account address for vote <VOTE_ACCOUNT>, expected <ACTIVE_STAKE_ADDRESS>, received <ACTIVATING_TRANSIENT_STAKE_ADDRESS>`.

Reply:
From the logs, it seems like the issue is not just with not withdrawing everything possible, it's also with trying to withdraw from transient stake accounts before the validator account.

These are the accounts available in the pool on devnet:
- Validator 1 active
- Validator 2 active
- Validator 3 active
- Validator 3 transient

This is the order that the bindings chose for withdrawing stake:
- Validator 1 active
- Validator 3 transient

As you see, the bindings chose to try to withdraw from validator 3 transient before trying validator 3 active.

Reply:
Also, when `useReserve` is set to true, it tries withdrawing all the balance from the reserve even if there aren't enough lamports in reserve to cover it fully.